imap_processing/ultra/l0/ultra_utils.py

```diff
@@ -0,0 +1,124 @@
+"""Contains data classes to support Ultra L0 processing."""
+from dataclasses import fields
+from enum import Enum
+from typing import NamedTuple
+
+from imap_processing.cdf.defaults import GlobalConstants
+
+
+class PacketProperties(NamedTuple):
+    """Class that represents properties of the ULTRA packet type."""
+
+    apid: int
+    width: int
+    block: int
+    len_array: int
+    mantissa_bit_length: int
+
+
+class UltraParams(Enum):
```

Comment:
I personally find the use of `.value.apid` and other properties kind of verbose in your other areas of code. Does this need to be an Enum, or could you just make each one be its own object in the main namespace?
```python
ULTRA_AUX = PacketProperties(...)
ULTRA_RATES = PacketProperties(...)
```